zebra/zebra_rnh.c

```diff
@@ -393,48 +395,25 @@ static struct route_entry *zebra_rnh_resolve_entry(vrf_id_t vrfid, int family,
 	if (!rn)
 		return NULL;
 
+	/* Unlock route node - we don't need to lock when walking the tree. */
+	route_unlock_node(rn);
+
 	/* When resolving nexthops, do not resolve via the default route unless
 	 * 'ip nht resolve-via-default' is configured.
 	 */
-	if ((type == RNH_NEXTHOP_TYPE)
-	    && (is_default_prefix(&rn->p)
-		&& !nh_resolve_via_default(rn->p.family)))
-		re = NULL;
-	else if ((type == RNH_IMPORT_CHECK_TYPE)
-		 && CHECK_FLAG(rnh->flags, ZEBRA_NHT_EXACT_MATCH)
-		 && !prefix_same(&nrn->p, &rn->p))
-		re = NULL;
-	else {
-		/* Identify appropriate route entry. */
-		RNODE_FOREACH_RE (rn, re) {
-			if (CHECK_FLAG(re->status, ROUTE_ENTRY_REMOVED))
-				continue;
-			if (!CHECK_FLAG(re->status, ROUTE_ENTRY_SELECTED_FIB))
-				continue;
+	if (((is_default_prefix(&rn->p)) ||
```

Comment:
This exception for default routes didn't exist before for `bgp network import-check`. Was this change intentional? Also, the comment above about "resolving nexthops" seems out of context in this function.

Reply:
The check for default routes isn't really needed for import-check.  Will remove and also fix the comment.